src/testRunner/unittests/tsserver/autoImportProvider.ts

```diff
@@ -0,0 +1,237 @@
+            // Simulate browsing typings files inside node_modules: no point creating auto import program
+            // for the InferredProject that gets created in there.
+            const { projectService, session } = setup([
+                angularFormsDts,
+                { path: angularFormsPackageJson.path, content: `{ "dependencies": { "@angular/core": "*" } }` },
+                { path: "/node_modules/@angular/core/package.json", content: `{ "typings": "./core.d.ts" }` },
+                { path: "/node_modules/@angular/core/core.d.ts", content: `export namespace angular {};` },
+            ]);
+
+            openFilesForSession([angularFormsDts], session);
+            checkNumberOfInferredProjects(projectService, 1);
+            checkNumberOfConfiguredProjects(projectService, 0);
+            assert.isUndefined(projectService
+                .getDefaultProjectForFile(angularFormsDts.path as server.NormalizedPath, /*ensureProject*/ true)!
+                .getLanguageService()
+                .getAutoImportProvider());
+        });
+
+        it("Auto-importable file is in inferred project until imported", () => {
+            const { projectService, session, updateFile } = setup([angularFormsDts, angularFormsPackageJson, tsconfig, packageJson, indexTs]);
+            checkNumberOfInferredProjects(projectService, 0);
+            openFilesForSession([angularFormsDts], session);
+            checkNumberOfInferredProjects(projectService, 1);
+            assert.equal(
+                projectService.getDefaultProjectForFile(angularFormsDts.path as server.NormalizedPath, /*ensureProject*/ true)?.projectKind,
+                server.ProjectKind.Inferred);
+
+            updateFile(indexTs.path, "import '@angular/forms'");
+            assert.equal(
+                projectService.getDefaultProjectForFile(angularFormsDts.path as server.NormalizedPath, /*ensureProject*/ true)?.projectKind,
+                server.ProjectKind.Configured);
+
+            assert.isUndefined(projectService.configuredProjects.get(tsconfig.path)!.getLanguageService().getAutoImportProvider());
+        });
+
+        it("Responds to package.json changes", () => {
+            const { projectService, session, host } = setup([
+                angularFormsDts,
+                angularFormsPackageJson,
+                tsconfig,
+                { path: "/package.json", content: "{}" },
+                indexTs
+            ]);
+
+            openFilesForSession([indexTs], session);
+            assert.isUndefined(projectService.configuredProjects.get(tsconfig.path)!.getLanguageService().getAutoImportProvider());
+
+            host.writeFile(packageJson.path, packageJson.content);
+            assert.ok(projectService.configuredProjects.get(tsconfig.path)!.getLanguageService().getAutoImportProvider());
+        });
+
+        it("Reuses autoImportProvider when program structure is unchanged", () => {

```

Comment:
Some tests are needed to ensure that autoImport project is closed and removed with project close